### ISSUE TITLE
LDAP Auth Fails with FreeIPA

### ISSUE THREAD

**Issue by Horvaticus** [snipe/snipe-it#2379]
#### Expected Behavior (or desired behavior if a feature request)

I am trying to authenticate LDAP users into the application using synced credentials from FreeIPA (OpenLDAP)
-----

#### Actual Behavior

When trying any LDAP account, I get the following error message in my laravel.log:

[2016-08-02 00:30:19] production.DEBUG: Local user USERNAME exists in database. Authenticating existing user against LDAP.
[2016-08-02 00:30:19] production.DEBUG: User USERNAME did not authenticate correctly against LDAP. Local user was not updated.
[2016-08-02 00:30:19] production.DEBUG: Authenticating user against database.
[2016-08-02 00:30:19] production.DEBUG: Local authentication failed.

I am able to log in with local user accounts.

I suspected it was the LDAP authentication query at first, but I have tried the following combinations to no avail:
uid=
uid="
uid
uid=uid
uid=samaccountname
samaccountname=
samaccountname=uid

(The last three wouldn't make any sense but I was desperate!)

I've also tried restarting the server / services / etc.

Here is my current configuration:
![image](https://cloud.githubusercontent.com/assets/4920785/17313292/298f2f1e-580f-11e6-86b2-2900daccf7ae.png)

Thank you for all of your hard work! I've been waiting for LDAP support for a long time and am really excited to use your tool.

-----

#### Please confirm you have done the following before posting your bug report:

- [x] I have enabled debug mode
- [x] I have read [checked the Common Issues page](http://docs.snipeitapp.com/common-issues.html)

-----
#### Please provide answers to these questions before posting your bug report:

- Version of Snipe-IT you're running
| Version v3.1-84-g139325d
- What OS and web server you're running Snipe-IT on
| Centos 7.2 x64, Apache
- What method you used to install Snipe-IT (install.sh, manual installation, docker, etc)
| Install.sh
- If you're getting an error in your browser, include that error
| No error in browser
- What specific Snipe-IT page you're on, and what specific element you're interacting with to trigger the error
| Log in page in an incognito window through Chrome
- If a stacktrace is provided in the error, include that too.
| Provided above
- Any errors that appear in your browser's error console.
| None
- Confirm whether the error is [reproduceable on the demo](https://snipeitapp.com/demo).
| No
- Include any additional information you can find in `app/storage/logs` and your webserver's logs.
| No relevant info from snipeIT.error.log or error_log from HTTPD
- Include what you've done so far in the installation, and if you got any error messages along the way.
| I have set up locations, configured LDAP, and then imported the users. No assets have been added.
- Indicate whether or not you've manually edited any data directly in the database
| No

**Comment by snipe** [comment]
We've had LDAP support for quite some time ;)

Do you have ldapsearch installed? (You don't need it for Snipe-IT, but it could be useful for troubleshooting.)

**Comment by Horvaticus** [comment]
I know! But last time I was working on a project that required asset management it wasn't quite ready :) 

And, yes, I do. Nothing strange comes up whenever I check the user I'm trying to authenticate as, or whenever I use the search filter I have configured. At the same time I verified that the service account is working properly too, although I suspect that import would have failed if that were the case

**Comment by snipe** [comment]
Did you get this sorted?

**Comment by Horvaticus** [comment]
I've figured this out!

So, I had to include the CNs in my Base Bind DN search - as Snipe was looking for my users at uid=username,dc=eng,dc=int,dc=systems. After inserting those it started finding the users!

I also had to use 'uid=' (without quotes) as my search parameter. The default entry "uid=" (with quotes) doesn't work.

Another thing I noticed is that the default value givenName won't work either unless it's changed to givenname. Suppose that makes sense since everything is case sensitive in Looney Tunes, but odd since it's capitalized in IPA.

Anyways, thanks for the time! I love your app :)
